apps/web/features/admin/rate-limit/rate-limit-form.tsx

```diff
@@ -0,0 +1,97 @@
+"use client";
+
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@repo/ui/components/form";
+import { Input } from "@repo/ui/components/input";
+import { toast } from "sonner";
+import { Button } from "@repo/ui/components/button";
+import { submitRateLimit } from "@/lib/actions/admin/rate-limit";
+import { useTransition } from "react";
+import type {
+  RateLimitResponse,
+  RateLimitResponseData,
+} from "@/types/rate-limit";
+import { TooltipWrapper } from "@/components/tooltip-wrapper";
+
+const FormSchema = z.object({
+  rate: z.coerce
+    .number()
+    .min(1, "Rate limit must not be less than 1")
+    .max(1000, "Rate limit must not be more than 1000"),
+});
+
+type Props = {
+  defaultValue: RateLimitResponseData | null;
+};
+
+export function RateLimitForm({ defaultValue }: Props) {
+  const [isPending, setTransition] = useTransition();
+  const form = useForm<z.infer<typeof FormSchema>>({
+    resolver: zodResolver(FormSchema),
+    defaultValues: {
+      rate: defaultValue?.limit || 0,
```

Comment:
Validation is min 1 and max 1000 but default value is 0

Reply:
it's better to set 1000 as the default value to prevent blocking the whole platform